pkg/virt-launcher/virtwrap/converter/vcpu/vcpu.go

```diff
@@ -454,18 +480,17 @@ func AdjustDomainForTopologyAndCPUSet(domain *api.Domain, vmi *v12.VirtualMachin
 		}
 	}
 
-	var emulatorThread uint32
+	var emulatorThreadsCPUSet string
 	if vmi.Spec.Domain.CPU.IsolateEmulatorThread {
-		emulatorThread, err = cpuPool.FitThread()
-		if err != nil {
-			e := fmt.Errorf("no CPU allocated for the emulation thread: %v", err)
-			log.Log.Reason(e).Error("failed to format emulation thread pin")
-			return e
+		vCPUs := hardware.GetNumberOfVCPUs(vmi.Spec.Domain.CPU)
```

Comment:
Hmm, I am not sure this will work with hotplug. Do we need the vCPUs?

Reply:
I think we addressed the CPU hotplug corner [here](https://github.com/kubevirt/kubevirt/pull/10757/files#diff-98b8339212e612cd92ce33649e492336892a3ed649f368b181aad9aa2854138cR484). I consulted with @enp0s3 and also checked it manually at the time (the original PR).